lib/dao.js

```diff
@@ -1183,8 +1191,9 @@ DataAccessObject._coerce = function (where) {
       }
     } else {
       if (val != null) {
-        if (!((operator === 'like' || operator === 'nlike') &&
-          val instanceof RegExp)) {
+        if (operator === 'regexp' && val instanceof RegExp) {
+          // do not coerce regex literals/objects
```

Comment:
Sure. With it in place, you don't have to coerce regexp in memory connector, right?